projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -404,6 +416,105 @@ private function parse_menu_item( $title ) {
 
 		return $item;
 	}
+
+	/**
+	 * Check if the given string is structured like a domain
+	 *
+	 * @TODO: Support deep nested domains like jp.eu.foo.something.something.wordpress.com
+	 * @param string $domain_name_value The string that will be checked.
+	 *
+	 * @return false|int
+	 */
+	private function is_valid_domain_name( $domain_name_value = '' ) {
+		return preg_match( '/^(http[s]?\:\/\/)?((\w+)\.)?(([\w-]+)?)(\.[\w-]+){1,2}$/', $domain_name_value );
+	}
+
+	/**
+	 * Generate a identifier based on menu or submenu slug
+	 *
+	 * @param string $key Menu or submenu slug.
+	 *
+	 * @return string
+	 */
+	private function generate_identifier( $key ) {
```

Comment:
I don't understand what this function does. Why cannot we simply return the given menu identifier / slug?

Reply:
The function is used to generate identifiers from slugs or custom identifiers (in case of menus) in a more consistent way and to filter out unique values (like domains). I'll update the docblock with more information, and maybe rename the function to better reflect the purpose of it.

Reply:
Still don't get why we cannot use the given menu identifier / slug?

Reply:
Or in other words, why `http://example.org`, `https://example.org`, `my-plugin.php`, `toplevel_page_my_plugin` are no valid identifiers?

Reply:
Ok, I think I got it more or less now. I guess this is only needed for submenus when a slug is used. In these case, we  want to clean and remove the slashes and domains from the identifier: e.g. `/stats/day/mydomain.blog` > `stats_day`.

While the domain certainly needs to be removed, I don't think we should worry about the slashes (`/stats/day` looks a good identifier to me).

With that in mind, I think we can simplify this function (and its call) to something like:

For menus:
```php
'identifier' => $menu_item[5],
```

For submenus:
```php
'identifier' => str_replace( '/' . ( new Status() )->get_site_suffix(), '', $url ),
```

If the above produces identifiers that don't match the existing Calypso events names, I'd suggest to maintain a larger mapping with hardcoded values. It certainly will be longer to maintain, but definitely more understandable for future developers (related: p7H4VZ-2Vx-p2).

Example:
```php
'/home' => `customer-home`, // https://github.com/Automattic/wp-calypso/blob/6e85729a75a71b83a9e96aeb378528c739ae71f1/client/my-sites/sidebar/index.jsx#L250
'/stats/day' => `stats`, // https://github.com/Automattic/wp-calypso/blob/6e85729a75a71b83a9e96aeb378528c739ae71f1/client/my-sites/sidebar/index.jsx#L218
'/posts' => `site`, // https://github.com/Automattic/wp-calypso/blob/6e85729a75a71b83a9e96aeb378528c739ae71f1/client/my-sites/sidebar/site-menu.jsx#L128
```